### ISSUE TITLE
submitAndWaitForTransaction fails for transactions with no effects

### ISSUE THREAD

**Issue by oggy-** [digital-asset/daml#6169]
If a command results in a transaction with no effects (creates/archives), the `submitAndWaitForTransaction` call will fail. Example:

https://github.com/digital-asset/daml/pull/6168

**Comment by SamirTalwar-DA** [comment]
Digging into this, it appears to be an issue in the API server (or potentially the indexer). When querying the index, we don't distinguish between a transaction with no events and no transaction at all.

**Comment by oggy-** [comment]
@SamirTalwar just as a note, `submitAndWaitForTransactionTree` works fine (because I suppose it finds at least one event)

**Comment by SamirTalwar-DA** [comment]
Yes, the code follows the same logic, so that's the most likely explanation.

**Comment by mziolekda** [comment]
Closing, the issue was fixed subsequently
